### ISSUE TITLE
Assume Role is not supported

### ISSUE THREAD

**Issue by jerger** [envato/ami-spec#28]
As I understand your docu, the only way to test specs is to  provide amazons key & key secret.
This means, amazons assume role feature is not supported yet?

BR, jerger

**Comment by patrobinson** [comment]
Assume role is definitely supported and working!

**Comment by myoung34** [comment]
Definitely works. 

@jerger how would you run tests against an instance using secret key? This ssh's into it using the `--key-name` and `--key-file` params.

I don't see any references to secret keys

**Comment by patrobinson** [comment]
Sorry, I don't know where my brain was at, instance profiles are supported. Assuming a role is best done before you call `ami-spec` like so:

```
aws sts assume-role --role-arn ...
ami-spec ...
```

**Comment by jerger** [comment]
Okay. So let me rephrase my question:
1.) assume-role will work for creating a test-instance from ami?
2.) the parameters --key-name and --key-file has to point to a key residing in aws?
3.) the --ssh-user @ test-instance will get the --key-file injected - so server-spec tests can be executed the given ssh-credentials ?

My ami allready contains needed credentials - so step 2. - 3. is not necessary to me ...

**Comment by patrobinson** [comment]
HI Jerger,

> 1.) assume-role will work for creating a test-instance from ami?

Yes, as listed above

> 2.) the parameters --key-name and --key-file has to point to a key residing in aws?

Yes

> 3.) the --ssh-user @ test-instance will get the --key-file injected - so server-spec tests can be executed 

Correct.

You could always just set `--key-name` to an existing key, but then ignore it and point `--key-file` at your existing SSH key. Unfortunately at the moment `--key-name` is a required parameter, although that's not strictly necessary.

**Comment by jerger** [comment]
Great - thanx for clarifying :)
